### DISCUSSION TITLE
Is there a way to manually rollback individual repos in multi sources

### DISCUSSION THREAD

**Discussion by carlosrodfern** [argoproj/argo-cd#17720] [Q&A]
When using helm, if there was a connectivity issue, for example, the helm release can get stuck in "preparing...". If you try to reupgrade, helm says "there is an upgrade in progress". The only way to get it out of that state is with a rollback.

ArgoCD doesn't support rollback in the UI, but then how can you get out of that stuck preparing upgrade state, if you can't use the helm command independently (helm doesn't even see the releases), and argo can't either?

**Comment by jgwest** [comment]
@carlosrodfern my understanding is that Helm hooks are mapped to the corresponding Argo CD hook, and thus use Argo CD's behaviour, rather than Helm's behaviour (which yes, can cause compatibility issues in some cases). More info here: https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks

So, for example, cancelling an Argo CD sync within Argo CD would interrupt any active hooks.

Likewise see the comment there, re `'install' vs 'upgrade' vs 'sync'`.

**Comment by carlosrodfern** [comment]
@jgwest Thank you for looking into this. This was more like a hypothetical question. We have an app being deployed currently with helm cli, but want to migrate it to be managed by ArgoCD. One of the problems we had sometimes was what I described in the OP.  When a release is stuck in the "preparing upgrade" state because of interrupted upgrades, any subsequent `helm upgrade` fails with a message like "there is another one in progress" We had a workaround with helm cli itself, by doing a `helm rollback` to get it out of that state, however, I was wondering how ArgoCD handles those situations taking into account we can't rollback in multi-source scenarios. Helm hooks are not in scope in this case.
